### PR TITLE
Introduce Gitian for reproducible builds

### PR DESCRIPTION
**Pull request by alpe**
This PR introduces [Gitian](https://gitian.org) in order to create reproducible builds. Gitian is also used by [Cosmos Gaia](https://github.com/cosmos/gaia) and this PR is very much based on their shared code and configuration examples. Thanks a lot for sharing!

See for further descriptions: 
* https://medium.com/tendermint/reproducible-builds-8c2eebb9a486 
* https://github.com/cosmos/gaia/blob/master/docs/reproducible-builds.md 

This PR includes

* Gitian descriptor 
* Build script
* TravisCI integration to build the bnsd binary with Gitian
* TravisCI to upload the binary to Github releases when tagged
* My developer key

Part of https://github.com/iov-one/ops/issues/183

### CODE REVIEW

File: contrib/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* Build, sign and upload
+replace `user@example.com` with the GPG identity you want to sign the report with
+
+```sh
+./contrib/gitian-build.sh -c -s user@example.com linux
+```
+
+## GPG
+
+# Generate key
```
Comment (husio) [inline contrib/README.md]: This and other sub paragraphs should be part of `## GPG`. Use heading level higher than 2.

File: contrib/gitian-build.sh [hunk 2]
```diff
@@ -0,0 +1,199 @@
+#!/bin/bash
+
+# This file was copied from the https://github.com/cosmos/gaia project with minor changes.
+
+# symbol prefixes:
+# g_ -> global
+# l_ - local variable
+# f_ -> function
```
Comment (husio) [inline contrib/gitian-build.sh]: I have no idea where they got it from. This is not vimscript. There are proper ways of doing this in shell. Using made up prefixes is questionable.

Comment (alpe) [reply contrib/gitian-build.sh]: 👍 got your point but let's keep the diff to a minimum

**Comment by alpe** [comment]
I will do another PR to apply changes for the gitian master on osx that require the host to run a apt-get proxy. Merging this now.

**Pull request merged**